homeassistant/components/rest/binary_sensor.py

```diff
@@ -149,24 +149,28 @@ def _update_from_rest_data(self) -> None:
             self._attr_is_on = False
             return
 
-        response = self.rest.data
+        response = self.rest.data_without_xml()
 
         raw_value = response
 
-        if self._value_template is not None:
+        if response is not None and self._value_template is not None:
             response = self._value_template.async_render_with_possible_json_value(
-                self.rest.data, False
+                response, False
             )
 
         try:
-            self._attr_is_on = bool(int(response))
+            if raw_value != self.rest.data:
```

Comment:
Is this really correct, it's causing the tests to fail, right?

I think it's better to modify `data_without_xml` to let the `ExpatError` bubble up and catch it in this function as well as in `RestSensor` and deal with it there.

Reply:
Seems to be failing on `test_setup_minimum_payload_template` which is an existing test - but that's beside the point, let me try my hand at implementing your idea. I will ping back once I'm done.

Reply:
Ok, looks fine now, submitting